Asignment7/Assignment7.py

```diff
@@ -1057,6 +1010,77 @@ def main():
 
     file_html.close()
 
+    return
+
+
+def main():
+
+    """loading the data"""
+    db_user = "root"
+    db_pass = "root"
+    db_host = "localhost"
+    db_database = "baseball"
+    connect_string = f"mariadb+mariadbconnector://{db_user}:{db_pass}@{db_host}/{db_database}"
+
+    sql_engine = sqlalchemy.create_engine(connect_string)
+
+    query = """
+                    SELECT * FROM baseball_game_stats
+                    """
+
+    df = pandas.read_sql_query(query, sql_engine)
+
+    predictors = ['HT_BA',
+                  'AT_BA',
+                  'HT_home_runs_per_hit',
+                  'AT_home_runs_per_hit',
+                  'HT_PA',
+                  'AT_PA',
+                  'HT_BABIP',
+                  'AT_BABIP',
+                  'HT_BAIP',
+                  'AT_BAIP',
+                  'HT_K_to_BB',
+                  'AT_K_to_BB',
+                  'HT_BB_to_K',
+                  'AT_BB_to_K',
+                  'HT_OBP',
+                  'AT_OBP',
+                  'HT_HCP',
+                  'AT_HCP',
+                  'HT_SP_WHIP',
+                  'AT_SP_WHIP',
+                  'HT_SP_K_per_9inn',
+                  'AT_SP_K_per_9inn',
+                  'HT_SP_BB_per_9_inn',
+                  'AT_SP_BB_per_9_inn',
+                  'HT_SP_H_per_9_inn',
+                  'AT_SP_H_per_9_inn']
+
+    response = "HT_Wins"
+
+    Name = "Baseball"
+
+    # getting all the stats
+
+    get_all_tables(df, predictors, response, Name)
+
+    """Splitting the data to test and train and applying a decision tree model """
+
+    X = df[predictors]
+
+    y=df[response]
+
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.33, random_state = 42)
+
+    model = DecisionTreeClassifier(criterion='gini', max_depth=3, random_state=0)
+
+    # fit the model
+    model.fit(X_train, y_train)
+
+    y_pred = model.predict(X_test)
+
+    print('Model accuracy score with criterion gini index: {0:0.4f}'.format(accuracy_score(y_test, y_pred)))
 
 if __name__ == "__main__":
```

Comment:
I'm unable to run the code but the functionalities from hw4 and midterm look good.